pgjdbc/src/main/java/org/postgresql/util/NumberParser.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright (c) 2023, PostgreSQL Global Development Group
+ * See the LICENSE file in the project root for more information.
+ */
+
+package org.postgresql.util;
+
+/**
+ * Optimised byte[] to number parser.
+ */
+public class NumberParser {
+  private static final NumberFormatException FAST_NUMBER_FAILED = new NumberFormatException() {
+    @Override
+    public Throwable fillInStackTrace() {
+      return this;
+    }
+  };
+
+  private static final long MAX_LONG_DIV_TEN = Long.MAX_VALUE / 10;
+
+  /**
+   * Optimised byte[] to number parser. This code does not handle null values, so the caller must do
+   * checkResultSet and handle null values prior to calling this function. Fraction part is
+   * discarded.
+   *
+   * @param bytes integer represented as a sequence of ASCII bytes
+   * @return The parsed number.
+   * @throws NumberFormatException If the number is invalid or the out of range for fast parsing.
+   *                               The value must then be parsed by another (less optimised) method.
+   */
+  public static long getFastLong(byte[] bytes, long minVal, long maxVal) throws NumberFormatException {
+    int len = bytes.length;
+    if (len == 0) {
+      throw FAST_NUMBER_FAILED;
+    }
+
+    boolean neg = bytes[0] == '-';
+
+    long val = 0;
+    int start = neg ? 1 : 0;
+    while (start < len) {
+      byte b = bytes[start++];
+      if (b < '0' || b > '9') {
+        if (b == '.') {
+          if (neg && len == 2 || !neg && len == 1) {
+            // we have to check that string is not "." or "-."
+            throw FAST_NUMBER_FAILED;
+          }
+          while (start < len) {
```

Comment:
Can we add a comment here to explain that we are reading the rest of the string to see if there are any more invalid bytes?

Reply:
Yes, sure, I added comment